File: ALG_lista2/ex4.py

def calcularDigitoVerificador(arr):
    soma1 = 0
    for i in range(0, 9):
        soma1 += arr[i] * (10 - i)
    
    if (soma1 % 11) < 2:
        arr[9] = 0
    else:
        arr[9] = 11 - (soma1 % 11)
    
    soma2 = 0
    for i in range(0, 10):
        soma2 += arr[i] * (11 - i)
    
    if (soma2 % 11) < 2:
        arr[10] = 0
    else:
        arr[10] = 11 - (soma2 % 11)

def imprimirCPF(arr):
    parte1 = ''.join(str(arr[i]) for i in range(9)) # List comprehension
    print(f"{parte1}-{arr[9]}{arr[10]}") 


if __name__ == "__main__":
    arr = [0] * 11

    entrada = input("Digite os 9 primeiros numeros de seu CPF:\n")

    if len(entrada) != 9 or not entrada.isdigit():    # .isdigit() -> verifica se todos os caracteres sao numeros
        print("Erro: Digite apenas 9 digitos numericos.")
    else:
        for i in range(0, 9):
            arr[i] = int(entrada[i])
    
        calcularDigitoVerificador(arr)
        imprimirCPF(arr)

File: ALG_lista2/ex4.c
#include <stdio.h>
#include <string.h>

void calcularDigitoVerificador(int arr[]){
    int soma1 = 0;
    for(int i = 0; i < 9; i++){
        soma1 += arr[i] * (10 - i);
    }

    if((soma1 % 11) < 2){
        arr[9] = 0;
    } else {
        arr[9] = 11 - (soma1 % 11);
    }

    int soma2 = 0;

    for(int i = 0; i < 10; i++){
        soma2 += arr[i] * (11 - i);
    }

    if((soma2 % 11) < 2){
        arr[10] = 0;
    } else {
        arr[10] = 11 - (soma2 % 11);
    }   
    
}

void imprimirCPF(int arr[]){
    for(int i = 0; i < 9; i++){
        printf("%d", arr[i]);
    }
    printf("-%d%d", arr[9], arr[10]);
}

int main(){

    int arr[11];
    char input[20];

    printf("Digite os 9 primeiros numeros de seu CPF:\n");
    scanf("%19s", input);

    if(strlen(input) != 9){
        printf("Erro: Digite exatamente 9 digitos.\n");
        return 1;
    }
    
    for(int i = 0; i < 9; i++){
        if(input[i] < '0' || input[i] > '9'){
            printf("Erro: Entrada invalida!\n");
            return 1;
        }
        arr[i] = input[i] - '0'; // conversao '4' - '0' -> 52 - 48 -> 4
    }

    calcularDigitoVerificador(arr);
    imprimirCPF(arr);

    return 0;
}